src/details/ArborX_DetailsExpandHalfToFull.hpp

```diff
@@ -0,0 +1,67 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_DETAILS_EXPAND_HALF_TO_FULL_HPP
+#define ARBORX_DETAILS_EXPAND_HALF_TO_FULL_HPP
+
+#include <ArborX_DetailsKokkosExtViewHelpers.hpp>
+#include <ArborX_DetailsUtils.hpp>
+
+#include <Kokkos_Core.hpp>
+
+namespace ArborX::Details
+{
+
+template <class ExecutionSpace, class Offsets, class Indices>
+void expandHalfToFull(ExecutionSpace const &space, Offsets &offsets,
```

Comment:
Just a note to myself: may want to explore a version of the function where we construct an array of pairs `(i, j)` and `(j, i)`, sort it, and then construct new indices and offsets.